### DISCUSSION TITLE
How to update Select options when another form value changes?

### DISCUSSION THREAD

**Discussion by cehenriques** [ant-design/ant-design#41316] [Q&A]
Hi,

I'm using a Form component with 2 items: a Radio and a Select. When the Radio value changes I want to update the list of options of the Select. How can I do it?
I tried putting the options on react state, but then if I update the state value on the onChange event of the Radio, even though it does what I'm looking for, it throws an error on the console saying: "Cannot update during an existing state transition (such as within `render`). Render methods should be a pure function of props and state."

**Comment by Yuiai01** [comment]
Hello, you can switch between select options based on the change of values in the Radio onChange method, like this https://codesandbox.io/s/biao-dan-chi-cun-antd-5-3-1-forked-z9zdrx?from-embed=&file=/demo.tsx

**Reply by cehenriques** [reply]
Hi @Yuiai01 ,

Thanks for the example, I already tried that before. Now I realize that this works when the form is empty, the issue starts when I prefill the form with a record.
Please check this example:
https://codesandbox.io/s/unruffled-spence-rkiii2?file=/demo.tsx

**Reply by Yuiai01** [reply]
> Hi @Yuiai01 ,
> 
> Thanks for the example, I already tried that before. Now I realize that this works when the form is empty, the issue starts when I prefill the form with a record. Please check this example: https://codesandbox.io/s/unruffled-spence-rkiii2?file=/demo.tsx

Hi @cehenriques , I found that there may be a few problems in your example that may make it look invalid.
First, we should pass the initialization data through the Form component `initialValues` instead of using the `setFieldsValue` method of the form instance.
Second, there is no need to set the Radio value again in the onChange method of the Radio component.
I made some changes to your code, you can take a look https://codesandbox.io/s/recursing-ully-nv7f0p?file=/demo.tsx. 
I hope I can help you.

**Reply by cehenriques** [reply]
Awesome @Yuiai01 ! My issue was being caused by using the `setFieldsValue` on the form instance instead of the `initialValues` property.
Thanks a lot for your help, I really appreciate it.
